libs/ngx-dynamic-form-builder/src/lib/utils/dynamic-form-group.ts

```diff
@@ -406,115 +143,498 @@ export class DynamicFormGroup<TModel> extends FormGroup {
+            const formGroup = this._fb.group(classValidators);
+
+            // Add all controls to the form group
+            Object.keys(formGroup.controls).forEach(ctrlKey => {
+              dynamicFormGroup.addControl(ctrlKey, formGroup.controls[ctrlKey]);
+            });
+
+            // Add a value change listener to the group. on change, validate
+            dynamicFormGroup.valueChanges.subscribe(data => {
+              dynamicFormGroup.validate(undefined, this._validatorOptions);
+            });
+
+            formArray.controls.push(dynamicFormGroup);
+
+            // Recusrively set the object value
+            (formArray.controls[i] as DynamicFormGroup<any>).object = (this._object && objectArray && objectArray[i])
+              ? objectArray[i]
+              : {};
+          }
+          else {
+            // Create control
+            const controlValue = (this._object && objectArray && objectArray[i]) ? objectArray[i] : undefined;
+            const newFormControl = new FormControl(controlValue, formControl.validator);
+            newFormControl.setParent(this);
+
+            // Add the control to the FormArray
+            formArray.controls.push(newFormControl);
           }
         }
       }
+
+      // Handle Control
+      else {
+        const newObject = this._object ? this._object[key] : [];
+        this.controls[key].setValue(this._object && newObject ? newObject : undefined);
+      }
     });
   }
-  validateAllFormFields() {
-    Object.keys(this.controls).forEach(field => {
-      const control = this.get(field);
-      if (control instanceof FormControl) {
-        control.markAsTouched({ onlySelf: true });
-      } else {
-        if (control instanceof DynamicFormGroup) {
-          control.validateAllFormFields();
-        } else {
-          if (control instanceof FormArray) {
-            for (let i = 0; i < (control as FormArray).controls.length; i++) {
-              if ((control as FormArray).controls[i] instanceof FormControl) {
-                ((control as FormArray).controls[i] as FormControl).markAsTouched({ onlySelf: true });
-              } else {
-                if ((control as FormArray).controls[i] instanceof DynamicFormGroup) {
-                  ((control as FormArray).controls[i] as DynamicFormGroup<any>).validateAllFormFields();
+}
+
+
+export function getClassValidators<TModel>(factoryModel: ClassType<TModel>, fields: Dictionary, validatorOptions?: ValidatorOptions) {
+
+  // Get the validation rules from the object decorators
+  const allValidationMetadatas: ValidationMetadata[] = getFromContainer(MetadataStorage).getTargetValidationMetadatas(factoryModel, '');
+
+  // Get the validation rules for the validation group: https://github.com/typestack/class-validator#validation-groups
+  const validationGroupMetadatas: ValidationMetadata[] = getFromContainer(MetadataStorage).getTargetValidationMetadatas(
```

Comment:
Renamed to call out it's use with https://github.com/typestack/class-validator#validation-groups